SPHINXsys/src/shared/particle_dynamics/dissipation_dynamics/particle_dynamics_dissipation.hpp

```diff
@@ -427,6 +427,55 @@ namespace SPH
 		}
 	}
 	//=================================================================================================//
+	template <typename VariableType>
+	DampingPairwiseFromShell<VariableType>::
+		DampingPairwiseFromShell(BaseContactRelation &contact_relation, const std::string &variable_name, Real eta)
+		: LocalDynamics(contact_relation.sph_body_),
+		  DataDelegateContact<BaseParticles, ShellParticles>(contact_relation),
+		  eta_(eta), Vol_(particles_->Vol_), mass_(particles_->mass_),
+		  variable_(*particles_->getVariableByName<VariableType>(variable_name))
+	{
+		for (size_t k = 0; k != contact_particles_.size(); ++k)
+		{
+			shell_variable_.push_back(contact_particles_[k]->template getVariableByName<VariableType>(variable_name));
+		}
+	}
+	//=================================================================================================//
+	template <typename VariableType>
+	void DampingPairwiseFromShell<VariableType>::interaction(size_t index_i, Real dt)
+	{
+		Real Vol_i = Vol_[index_i];
+		Real mass_i = mass_[index_i];
+		VariableType &variable_i = variable_[index_i];
+
+		std::array<Real, MaximumNeighborhoodSize> parameter_b;
```

Comment:
These arrays can cause memory errors for multiresolution, where the neighborhood size is larger than the MaximumNeighborhoodSize variable. I think we mentioned this once but I don't remember what our conclusion was with prof. Hu. It's not very efficient either, perhaps @FabienPean-Virtonomy has a better solution?

Reply:
Yes it happened I had some problems there in the past, it was more related to some system resolution compared to particle spacing. There will be issues for multiresolution definitely. 

Two simpler options I see:
* The simpler fix would be using a std::vector with reserved memory initial memory (by calculating max neighborhood size)
* Using a couple array + vector and adjusting the loops to fill the vector once the array is filled. 

Otherwise other solutions could involve a different container outside STL (but require adding dependency), thread local vector (need to be confident about concurrency)